test/mutuable_data.js

```diff
@@ -7,32 +7,171 @@ const createAuthenticatedTestApp = h.createAuthenticatedTestApp;
+      return should(app.mutableData.newPublic(TEST_NAME_PUBLIC, TAG_TYPE_INVALID)).be.rejected();
+    });
+
+    it('create custom private with invalid tag value', () => {
+      return should(app.mutableData.newPrivate(TEST_NAME_PRIVATE, TAG_TYPE_INVALID)).be.rejected();
+    });
+
+    it('create custom public with invalid name', () => {
+      return should(app.mutableData.newPublic(TEST_NAME_INVALID, TAG_TYPE)).be.rejected();
+    });
+
+    it('create custom private with invalid name', () => {
+      return should(app.mutableData.newPrivate(TEST_NAME_INVALID, TAG_TYPE)).be.rejected();
+    });
+  })
+
+  describe('MutableData info', () => {
+    it('create random public entry and read its name', () =>
+        app.mutableData.newRandomPublic(TAG_TYPE)
+            .then((m) => m.getNameAndTag())
+            .then((r) => {
+              should(r.name).not.be.undefined();
+              should(r.tag).equal(TAG_TYPE);
+            })
+    );
+
+    it('create random private entry and read its name', () =>
+        app.mutableData.newRandomPrivate(TAG_TYPE)
+            .then((m) => m.getNameAndTag())
+            .then((r) => {
+              should(r.name).not.be.undefined();
+              should(r.tag).equal(TAG_TYPE);
+            })
+    );
+
+    it('create custom public entry and read its name', () =>
+        app.mutableData.newPublic(TEST_NAME_PUBLIC, TAG_TYPE)
+            .then((m) => m.getNameAndTag())
+            .then((r) => {
+              should(r.name).not.be.undefined();
+              // test XOR_NAME generation algorithm applied to the name provided???
+              should(r.name).have.length(TEST_NAME_PUBLIC.length);
+              should(r.tag).equal(TAG_TYPE);
+            })
+    );
+
+    it('create custom private entry and read its name', () =>
+        app.mutableData.newPrivate(TEST_NAME_PRIVATE, TAG_TYPE)
+            .then((m) => m.getNameAndTag())
+            .then((r) => {
+              should(r.name).not.be.undefined();
+              // test XOR_NAME generation algorithm applied to the name provided???
+              should(r.name).have.length(TEST_NAME_PUBLIC.length);
+              should(r.tag).equal(TAG_TYPE);
+            })
+    );
+
+    it.only('mdata version', () => {
+      return app.mutableData.newRandomPrivate(TAG_TYPE)
+          .then((m) => m.put().then(() => m.getVersion()))
+          .then((version) => {
+            should(version).equal(0);
+            // test that after a change in mdata (not in the entries) version is incremented
+          })
+
+    });
+  });
+
+  describe('Entries', () => {
+      it('get non-existing key', () => app.mutableData.newRandomPrivate(TAG_TYPE)
+          .then((md) => md.get('_non-existing-key'))
+          .then((value) => {
+            console.log("value:", value);
+            should(value).not.be.undefined();
+          })
+      );
+
+      it('insert and get a single entry', () => {
```

Comment:
please don't submit empty tests, they will be interpreted as "passing" which is misleading. Please instead throw an exception (like `throw new Error('Not Implemented')`) and for them to not be blocking your merges please mark them as to be skipped by using `it.skip(`.